### PR TITLE
refactor filters and conditional rendering

### PR DESCRIPTION
**Pull request by connectbo**

### CODE REVIEW

File: src/contexts/ServerContext.js [hunk 1]
```diff
@@ -91,36 +91,111 @@ export const ServerProvider = ({ children }) => {
...
-        });
+                setIsLoadingRescContext(false);
+            }).catch(() => {
+                setRescContext(undefined)
+                setIsLoadingRescContext(false);
+            });
+        }
+        else {
+            let filteredResults = {};
+            let resc_name_query = `SELECT RESC_NAME,RESC_TYPE_NAME,RESC_ZONE_NAME,RESC_VAULT_PATH,RESC_LOC,RESC_INFO, RESC_FREE_SPACE, RESC_COMMENT,RESC_STATUS,RESC_CONTEXT,RESC_PARENT,RESC_ID WHERE RESC_NAME != 'bundleResc' AND RESC_NAME LIKE '%${name}%'`
```
Comment (trel) [inline src/contexts/ServerContext.js:159]: can we use base_query here and just add to it?   same a few lines lower... keep a single source of truth for this set of columns...

File: src/views/resources/ResourceListView.js [hunk 2]
```diff
@@ -68,6 +68,7 @@ export const ResourceListView = () => {
         loadResource((currPage - 1) * perPage, perPage, filterRescName, order, orderBy);
     }, [loadResource, currPage, perPage, filterRescName, order, orderBy])
 
+
```
Comment (trel) [inline src/views/resources/ResourceListView.js:71]: can remove this once we're happy with the functionality...

File: src/views/resources/ResourceListView.js [hunk 3]
```diff
@@ -146,6 +147,7 @@ export const ResourceListView = () => {
         <Fragment>
             {isLoadingRescContext ? <LinearProgress /> : <div style={{ height: '4px'}}/>}
             <br />
+            {console.log(rescContext)}
```
Comment (trel) [inline src/views/resources/ResourceListView.js:150]: can remove this once we're happy with the functionality...

File: src/contexts/ServerContext.js [hunk 4]
```diff
@@ -91,36 +91,111 @@ export const ServerProvider = ({ children }) => {
         });
     }
 
+    // const sortResource = (offset, limit, order, orderBy) => {
+    //     resourceSortRemoveDuplicatesHelper(rescContext._embedded, offset, limit, order, orderBy)
+    // }
```
Comment (trel) [inline src/contexts/ServerContext.js:96]: unneeded?

**Review by trel — COMMENTED** [review]
looks like it works to me.

only thing is that when filter and there are zero results... might be nice to have a little default thing that says 'No results found'.   Definitely not a blocker.

**Comment by connectbo** [comment]
Sorry just saw your comments, too many meetings today :) 

Ok, sounds like the functionality is working, will do some cleaning. I can also create a separate issue for displaying 'No results found', feels like other pages (users, groups) should have this too?

**Comment by trel** [comment]
yes, a new issue for 'no results'... thanks.

**Comment by connectbo** [comment]
Alright. This pr is ready for review:
- filtering should be working as expected
- added 'no results for [text]'
- optimized the tableview rendering logic

**Comment by trel** [comment]
it works 

the 'No results found for []' could be in a 'table' row (white background) - i didn't expect it to be 'lost' in the background grey.

but it's not a blocker if you feel strongly.

**Comment by connectbo** [comment]
Alright, I refactored how 'No results found' renders on each page (normal tableview & editing user/group), also changed the textfield placeholder on resource page to 'Filter by name or hostname'. Ready for your review.

image
image

**Comment by trel** [comment]
i like it a lot.

I see two tiny inconsistencies...   The column name and the word in the 'Filter' box...

Screen Shot 2021-09-16 at 1 49 26 PM

Screen Shot 2021-09-16 at 1 49 32 PM

I propose... 'User Name' and 'Group Name' as the 'correct' way to display these.   We'll leave Hostname out of it for now :)

**Comment by connectbo** [comment]
Ahh yes, right. These should be consistent, will change both.

**Comment by trel** [comment]
we're really close now...

please make the new field say 'Group Name' too.
Screen Shot 2021-09-16 at 9 06 47 PM

and this one say 'User Name'
Screen Shot 2021-09-16 at 9 08 09 PM

promise that's it!

**Comment by connectbo** [comment]
Ohh right, I tried to look for 'Username' and 'GroupName' in my code and found nothing, should have used case-insensitive search. I will get this correct.

**Comment by trel** [comment]
confirmed - please # both!

**Comment by connectbo** [comment]
yep, added both!

**Pull request merged**